package com.rodrigodev.xgen4j.test.code.codeIdAndNameAreGeneratedFromErrorNameWhenNotSpecified_specialCases;



/**
 * Autogenerated by XGen4J on January 1, 0001.
 */
public class Root__Name__9__Name3Exception extends RuntimeException {

    final private ErrorCode errorCode;

    protected Root__Name__9__Name3Exception(ErrorCode errorCode, String message) {
        super(message);
        this.errorCode = errorCode;
    }

    protected Root__Name__9__Name3Exception(ErrorCode errorCode, String message, Throwable cause) {
        super(message, cause);
        this.errorCode = errorCode;
    }

    public ErrorCode errorCode() {
        return errorCode;
    }

    public abstract static class ExceptionType {

        protected ExceptionType() {
        }

        protected abstract Root__Name__9__Name3Exception createException(ErrorCode errorCode, String message);

        protected abstract Root__Name__9__Name3Exception createException(ErrorCode errorCode, String message, Throwable cause);
    }
}
